libs/community/langchain_community/retrievers/bm25s.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Iterable, List, Optional
+
+from langchain_core.callbacks import CallbackManagerForRetrieverRun
+from langchain_core.documents import Document
+from langchain_core.pydantic_v1 import Field
+from langchain_core.retrievers import BaseRetriever
+
+
+class BM25SRetriever(BaseRetriever):
+    """`BM25S` retriever."""
+
+    retriever: Any
+    """ BM25 retriever."""
+    docs: List[Document] = Field(repr=False)
```

Comment:
A copy of docs should not be retained after the index is created

Reply:
Same comment as above